src/NexusRpc/Handlers/OperationContext.cs

```diff
@@ -0,0 +1,51 @@
+using System.Collections.Generic;
+using System.Threading;
+
+namespace NexusRpc.Handlers
+{
+    /// <summary>
+    /// Base class for contexts provided on different operation calls.
+    /// </summary>
+    /// <param name="Service">Service name.</param>
+    /// <param name="Operation">Operation name.</param>
+    /// <param name="CancellationToken">Cancellation token for when this specific call is
+    /// canceled (not to be confused with operation cancellation).</param>
+    public abstract record OperationContext(
```

Comment:
You may want to consider adding a way to check the deadline of the operation. Similar to Go's `context.Deadline ` so users can know how much longer their operation can run for.

Reply:
Thanks for catching that @Quinn-With-Two-Ns, it is indeed a recent addition that Java got.